sound/soc/sof/pcm.c

```diff
@@ -619,6 +619,33 @@ static int sof_pcm_new(struct snd_soc_component *component,
 	return 0;
 }
 
+static int ssp_dai_config_pcm_params_match(struct snd_sof_dev *sdev, const char* name,
+					   struct snd_pcm_hw_params *params)
+{
+	struct sof_ipc_dai_config *config;
+	struct snd_sof_dai *dai;
+	int number_configs;
+	int ret = -EINVAL;
+	int i;
+
+	/* we might have several dai's associated with same link, need to loop all */
+	list_for_each_entry(dai, &sdev->dai_list, list) {
+		if (dai->name && (strcmp(name, dai->name) == 0)) {
+			number_configs = dai->number_configs;
+			for (i = 0; i < number_configs; i++) {
+				config = &dai->dai_config[i];
+				if (config->ssp.fsync_rate == params_rate(params)) {
+					dev_dbg(sdev->dev, "found DAI config pcm params match at index %d\n", i);
+					dai->current_config = i;
+					ret = 0;
+				}
+			}
+		}
+	}
+
+	return ret;
```

Comment:
You dont do anything with the return value? nor do you modofy current_config in case no match is found? It is left as default and so what is the outcome when no match is found?

Reply:
it is using default

Reply:
@ranj063 so the idea is that in the future I can call this function: if returns NULL I can search for something else, and so on.